eclass/kernel-2.eclass

```diff
@@ -1435,22 +1435,19 @@ unipatch() {
 # pulled from linux-info
 
 getfilevar() {
-	local workingdir basefname basedname xarch=$(tc-arch-kernel)
+	local basefname basedname xarch=$(tc-arch-kernel)
 
 	if [[ -z ${1} ]] && [[ ! -f ${2} ]]; then
 		echo -e "\n"
 		eerror "getfilevar requires 2 variables, with the second a valid file."
 		eerror "   getfilevar <VARIABLE> <CONFIGFILE>"
 	else
-		workingdir=${PWD}
 		basefname=$(basename ${2})
 		basedname=$(dirname ${2})
 		unset ARCH
 
-		cd ${basedname}
-		echo -e "include ${basefname}\ne:\n\t@echo \$(${1})" | \
-			make ${BUILD_FIXES} -s -f - e 2>/dev/null
-		cd ${workingdir}
+		echo -e "include ${basefname}\ne:\n\t@echo \$(${1})" |
+		emake -C "${basedname}" ${BUILD_FIXES} -s -f - e 2>/dev/null
```

Comment:
Perhaps keep indentation here, to indicate the continuation (I note dropping possibly redundant backslash)?